Introduce the four-color pen

octareenroon91 opened the pull request:
Add a fancy pen that lets you choose between black, red, green, and blue ink.

:cl:
rscadd: Add the four-color pen, which writes in black, red, green, and blue.
rscadd: Adds two such pens to the Bureaucracy supply pack.
/:cl:

octareenroon91 committed:
Introduce the four-color pen

Add a fancy pen that lets you choose between black, red, green, and blue ink.

Anonus commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,31 @@
+		else
+			colour = "black"
+	user << "<span class='notice'>[src] will now write in [src.colour].</span>"
+	//user << "<span class='notice'>[src] will now write in <span style='color=[colour]'>[colour]</span>.</span>" // If anyone feels lucky
```
not span style, font style!

optimumtact replied:
also remove this dead comment

octareenroon91 replied:
That will make the colored feedback work?

Anonus replied:
Yes it will.

MrStonedOne commented:
New content, Please make a changelog.

You can include a changelog in the body of the pr now: https://tgstation13.org/phpBB/viewtopic.php?f=5&t=5157

Iamgoofball commented:
why not make it allow a color hex?

octareenroon91 committed:
Make four-color pen available through cargo

Adds two of /obj/item/weapon/pen/fourcolor to the Bureaucracy crate.

octareenroon91 commented:
I wasn't sure how hard it was to pull up the color hex interface.
Also the four-color pen is iconic, and should exist whether or not a more expansive pen does.

YotaXP commented:
You can bring up the color picker with `input(...)as color`, but I agree that 4-color is better.

WJohn commented:
Where's the "awww" tag?

octareenroon91 commented:
What purpose would that serve?

WJohn commented:
There's so many adorable additions that should have "awww" tags.

Cheridan commented:
Too OP. This instantly invalidates all other pens! Think of the game's delicate balance.

RemieRichards commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,31 @@
+
+/obj/item/weapon/pen/fourcolor/New()
+	..()
+	colour = "black"
```
just set these by default

RemieRichards commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,31 @@
+/obj/item/weapon/pen/fourcolor/New()
+	..()
+	colour = "black"
+	desc = "It's a fancy four-color ink pen, set to [src.colour]."
```
set this by default, swapping ```[src.colour]``` here with "black" to match the line above.

RemieRichards commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,31 @@
+			colour = "blue"
+		else
+			colour = "black"
+	user << "<span class='notice'>[src] will now write in [src.colour].</span>"
```
remove ```src.```

RemieRichards commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,31 @@
+			colour = "black"
+	user << "<span class='notice'>[src] will now write in [src.colour].</span>"
+	//user << "<span class='notice'>[src] will now write in <span style='color=[colour]'>[colour]</span>.</span>" // If anyone feels lucky
+	desc = "It's a fancy four-color ink pen, set to [src.colour]."
```
remove ```src.```

RemieRichards commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,31 @@
+/obj/item/weapon/pen/fourcolor
+	desc = "It's a fancy four-color ink pen."
+	name = "four-color pen"
+	icon = 'icons/obj/bureaucracy.dmi'
```
this is inherited, remove this line

RemieRichards commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,31 @@
+	desc = "It's a fancy four-color ink pen."
+	name = "four-color pen"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "pen"
```
this is inherited, remove this line

octareenroon91 replied:
the invisible pen retains its duplicate icon_state, though

RemieRichards commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,31 @@
+	name = "four-color pen"
+	icon = 'icons/obj/bureaucracy.dmi'
+	icon_state = "pen"
+	item_state = "pen"
```
this is inherited, remove this line

octareenroon91 commented:
How about if there's only one and it spawns on the captain?

GunHog commented:
Pretty sure he was joking. Maybe.

octareenroon91 committed:
Four-color pen code improvement.

Take advantage of inheritance; use defaults instead of New()

optimumtact commented on code/modules/paperwork/pen.dm:
```diff
@@ -44,6 +44,24 @@
+		else
+			colour = "black"
+	user << "<span class='notice'>[src] will now write in [colour].</span>"
+	//user << "<span class='notice'>[src] will now write in <font style='color=[colour]'>[colour]</font>.</span>"
```
REMOVE IT

octareenroon91 committed:
remove commented-out code from pen.dm

fine

Cheridan commented:
@GunHog He actually tagged it, the absolute madman

duncathan commented:
I will not rest until bureaucracy is 100% CSGO balanced

YotaXP commented:
Librarian could use a buff.  How about letting their PDA start with one of these bad boys?

RemieRichards commented:
@YotaXP I dont think you realise how absurd you sound right now... the Librarian has been a main staple of  teams for a long time now.

octareenroon91 committed:
Typo in four-color pen feedback

Change message starting, "pen will now write..."
Now it starts grammatically correct: "The pen will now..."
